x-pack/plugins/observability_ai_assistant/server/routes/functions/route.ts

```diff
@@ -276,11 +276,92 @@ const setupKnowledgeBaseRoute = createObservabilityAIAssistantServerRoute({
   },
 });
 
+const functionGetDatasetInfoRoute = createObservabilityAIAssistantServerRoute({
+  endpoint: 'POST /internal/observability_ai_assistant/functions/get_dataset_info',
+  params: t.type({
+    body: t.type({
+      index: t.string,
+    }),
+  }),
+  options: {
+    tags: ['access:ai_assistant'],
+  },
+  handler: async (
+    resources
+  ): Promise<{
+    indices: string[];
+    fields: Array<{ name: string; description: string; type: string }>;
+  }> => {
+    const esClient = (await resources.context.core).elasticsearch.client.asCurrentUser;
+
+    const savedObjectsClient = (await resources.context.core).savedObjects.getClient();
+
+    const index = resources.params.body.index;
+
+    let indices: string[] = [];
+
+    try {
+      const body = await esClient.indices.resolveIndex({
```

Comment:
does this work for data streams? Asking because it doesn't seem to return any info about data streams. It also returns indices that do not match the pattern.

<img width="877" alt="CleanShot 2023-09-28 at 12 34 50@2x" src="https://github.com/elastic/kibana/assets/352732/1effb635-76d6-4538-91df-6d42251f7902">

Reply:
its using dataviews api to retreive the list of indices (and fields) so as long as that one supports data streams this should as well.

Reply:
Hmm, I couldn't get it to work - have you verified it works with data streams in your env? Maybe I'm missing something obvious.

Reply:
you are right, datastreams were missing and are now added.